conans/test/functional/toolchain/test_make.py

```diff
@@ -0,0 +1,306 @@
+import os
+import platform
+import textwrap
+import unittest
+
+from nose.plugins.attrib import attr
+
+from conans.test.utils.tools import TestClient
+
+from parameterized.parameterized import parameterized
+
+
+@attr("toolchain")
+class Base(unittest.TestCase):
+
+    conanfile = textwrap.dedent("""
+        from conans import ConanFile, MakeToolchain
+        class App(ConanFile):
+            settings = "os", "arch", "compiler", "build_type"
+            requires = "hello/0.1"
+            generators = "make"
+            options = {"shared": [True, False], "fPIC": [True, False]}
+            default_options = {"shared": False, "fPIC": True}
+            def toolchain(self):
+                tc = MakeToolchain(self)
+                tc.definitions["SOME_DEFINITION"] = "SomeValue"
+                return tc
+
+
+            def build(self):
+                self.run("make -C .. app")
```

Comment:
Great that you are simplifying and using it this way for this PR, to avoid needing to change the ``AutotoolsBuildEnvironment`` as we had to for CMake. If needed, we can think about new build-helpers later.